system-tests/tests/smoke/cre/por_test.go

```diff
@@ -95,40 +105,58 @@ func ExecutePoRTest(t *testing.T, testEnv *TestEnvironment, priceProvider PriceP
 		require.NoError(t, forwarderErr, "failed to find Forwarder address for chain %d", chainSelector)
 
-		uniqueWorkflowName := cfg.WorkflowName + "-" + bcOutput.BlockchainOutput.ChainID + "-" + uuid.New().String()[0:4] // e.g. 'por-workflow-1337-5f37_config'
-		configInput := &cre.ConfigureDataFeedsCacheInput{
-			CldEnv:                fullCldEnvOutput.Environment,
-			ChainSelector:         chainSelector,
-			FeedIDs:               []string{feedID},
-			Descriptions:          []string{"PoR test feed"},
-			DataFeedsCacheAddress: dataFeedsCacheAddress,
-			AdminAddress:          bcOutput.SethClient.MustGetRootKeyAddress(),
-			AllowedSenders:        []common.Address{forwarderAddress},
-			AllowedWorkflowNames:  []string{uniqueWorkflowName},
-			AllowedWorkflowOwners: []common.Address{bcOutput.SethClient.MustGetRootKeyAddress()},
+		if chainFamily == blockchain.FamilyTron {
+			dataFeedsCacheAddress, readBalancesAddress = deployAndConfigureTronContracts(t, testLogger, chainSelector, fullCldEnvOutput, workflowOwner, uniqueWorkflowName, feedID, forwarderAddress)
+
+			chainFamily = "evm"
+		} else {
+			workflowOwner = bcOutput.SethClient.MustGetRootKeyAddress()
+			testLogger.Info().Msgf("Deploying additional contracts to chain %d (%d)", chainID, chainSelector)
+			dfAddress, dfOutput, dfErr := crecontracts.DeployDataFeedsCacheContract(testLogger, chainSelector, fullCldEnvOutput)
+			require.NoError(t, dfErr, "failed to deploy Data Feeds Cache contract on chain %d", chainSelector)
+
+			rbAddress, rbOutput, rbErr := crecontracts.DeployReadBalancesContract(testLogger, chainSelector, fullCldEnvOutput)
+
+			require.NoError(t, rbErr, "failed to deploy Read Balances contract on chain %d", chainSelector)
+
+			dataFeedsCacheAddress = dfAddress
+			readBalancesAddress = rbAddress
+			crecontracts.MergeAllDataStores(fullCldEnvOutput, dfOutput, rbOutput)
+
+			testLogger.Info().Msgf("Configuring Data Feeds Cache contract...")
+
+			configInput := &cre.ConfigureDataFeedsCacheInput{
+				CldEnv:                fullCldEnvOutput.Environment,
+				ChainSelector:         chainSelector,
+				FeedIDs:               []string{feedID},
+				Descriptions:          []string{"PoR test feed"},
+				DataFeedsCacheAddress: dataFeedsCacheAddress,
+				AdminAddress:          bcOutput.SethClient.MustGetRootKeyAddress(),
+				AllowedSenders:        []common.Address{forwarderAddress},
+				AllowedWorkflowNames:  []string{uniqueWorkflowName},
+				AllowedWorkflowOwners: []common.Address{bcOutput.SethClient.MustGetRootKeyAddress()},
+			}
+			_, dfConfigErr := crecontracts.ConfigureDataFeedsCache(testLogger, configInput)
+			require.NoError(t, dfConfigErr, "failed to configure Data Feeds Cache contract")
+			testLogger.Info().Msg("Data Feeds Cache contract configured successfully.")
```

Comment:
could you wrap it in a function just like `deployAndConfigureTronContracts` so that we keep the same level of abstraction?